ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/MainActivity.kt

```diff
@@ -71,44 +74,96 @@ class MainActivity : ComponentActivity() {
 @Composable
 fun WearApp(watchRepository: WatchRepository) {
     WearAppTheme {
-        // The state for the ScalingLazyColumn in other screen is hoisted to this level, so
-        // the Scaffold can properly place the position indicator (also known as the
-        // scroll indicator). We use it for various other things (like hiding the time when
-        // the user is scrolling).
-        //
-        // This current solution works for one scrolling type Composable, that is, we only
-        // have one Composable capable of scrolling.
-        val scalingLazyListState: ScalingLazyListState = rememberScalingLazyListState()
+        // The state for any scrollable content in the composable screens is hoisted to this level,
+        // so the Scaffold can properly place the position indicator (also known as the
+        // scroll indicator).
+        // In our case, the watch list screen uses a ScalingLazyColumn and the watch details
+        // screen uses a Column with scrolling enabled.
+        // We also use the states for various other things (like hiding the time when the user is
+        // scrolling).
+        // ----------
+
+        // For all scrolling screens, we save their scrolling offset in a remembered map (as a
+        // ScrollableState), so when users return to the screen, they are at the same location as
+        // they left.
+        // We set the default to the watch list screen (ScalingLazyListState) since there is only
+        // one. For the watch detail screens (with many different variations based on the watch id),
+        // we add those dynamically to the map as they are needed (in the navigation section).
+        var currentScrollStateKey: String = remember { Screen.WatchList.route }
+        val scrollStates: MutableMap<String, ScrollableState> = remember {
```

Comment:
My concern about this is it looks like the kind of logic we expect ultimately every app with navigation to implement. I'd argue we should push to move this logic, and any logic for showing/hiding or scaling, offsettting the TimeText into Scaffold.

cc @JohnNichol

Reply:
+1

Reply:
Created a bug: https://issuetracker.google.com/issues/217446791

Reply:
@codingjeremy the use of the map looks fine to me. 

I don't think this logic should be moved into the Scaffold - the Scaffold doesn't know about navigation or app structure. If we start to move things into the Scaffold then it needs to start to expose all-sorts of the detail back via it's state to allow the application to manage things like transitions. We spent a lot of time with API Sponsors early on workout out whether the Scaffold knew about Vignettes, PositionIndicators, etc. and came to the conclusion that it made the Scaffold inflexible and difficult to use. 

It's a reasonable question to consider whether there is a component/components that contain a Vignette and provide this functionality for reuse in different application contexts.